test/StreamTest.php

```diff
@@ -246,6 +246,111 @@ public function testIsWritableReturnsFalseWhenStreamIsDetached()
         $this->assertFalse($stream->isWritable());
     }
 
+    public function testIsWritableReturnsTrueForWritableMemoryStream()
+    {
+        $stream = new Stream("php://temp", "r+b");
+        $this->assertTrue($stream->isWritable());
+    }
+
+    public function provideDataForIsWritable()
+    {
+        return [
+            ['a',   true,  true],
+            ['a+',  true,  true],
+            ['a+b', true,  true],
+            ['ab',  true,  true],
+            ['c',   true,  true],
+            ['c+',  true,  true],
+            ['c+b', true,  true],
+            ['cb',  true,  true],
+            ['r',   true,  false],
+            ['r+',  true,  true],
+            ['r+b', true,  true],
+            ['rb',  true,  false],
+            ['rw',  true,  true],
+            ['w',   true,  true],
+            ['w+',  true,  true],
+            ['w+b', true,  true],
+            ['wb',  true,  true],
+            ['x',   false, true],
+            ['x+',  false, true],
+            ['x+b', false, true],
+            ['xb',  false, true],
+        ];
+    }
+
+    private function findNonExistentTempName()
+    {
+        while (true) {
+            $tmpnam = sys_get_temp_dir() . DIRECTORY_SEPARATOR . 'diac' . uniqid();
```

Comment:
See docs - `tempnam` creates new file. If you open it with `x` mode, you get a warning:

```bash
PHP Warning:  fopen(/tmp/zrazAZD39N): failed to open stream: File exists in phar:///home/mat/bin/boris.phar/lib/Boris/EvalWorker.php(133) : eval()'d code on line 1
```